Automattic/wp-calypso
# Plans Redesign: use a table in PlansFeatures

This PR fixes #6381 by refactoring the PlansFeatures component to use a table, and also adds a top call to action. To see this in action start Calypso with ` ENABLE_FEATURES=manage/plan-features make run`

For a mobile view, I opted to add in a horizontal scroll for the first iteration. We could collapse to a single column, but the problem is how we tab back and forth between the plan views. `/plans` already has a SectionNav, while the nux flow does not, and Jetpack has an additional monthly vs annual toggle.

Desktop:
<img width="996" alt="screen shot 2016-07-06 at 5 02 51 pm" src="https://cloud.githubusercontent.com/assets/1270189/16638442/2b4805f6-439c-11e6-917c-8607109b630b.png">

Mobile:
![screen shot 2016-07-07 at 4 48 57 pm](https://cloud.githubusercontent.com/assets/1270189/16673427/e8a9bbf0-4462-11e6-8c18-2431e6cc07a8.png)

Work Left:
- [x] Add mobile breakpoints.
- [x] fix the bottom border when one feature item is longer than the others. (Try displaying all plans together)
- [x] Update the placeholder and wire it up again

## Testing Instructions

- Start Calypso with `ENABLE_FEATURES=manage/plan-features make run`
- Navigate to http://calypso.localhost:3000/plans
- Select a wpcom site with a free plan
- Verify that the plans table looks OK
- Navigate to http://calypso.localhost:3000/plans
- Select a jetpack site
- Verify that the plans table looks OK
- Navigate to http://calpso.localhost:3000/start
- Verify that the plans table looks OK on the plans step

cc @rralian @apeatling @mtias @lamosty 

Test live: https://calypso.live/?branch=update/plans-table
